pymeasure/instruments/rohdeschwarz/fsl.py

```diff
@@ -257,3 +249,29 @@ def zoom(self, value):
                 passed it is interpreted as a frequency span.
             """
             self.write(f"FUNC:ZOOM {value}; *WAI")
+
+    # Channels -----------------------------------------------------------------
+
+    def create_channel(self, channel_type, channel_name):
+        strict_discrete_set(channel_type, ["PNOISE", "SANALYZER"])
+        self.write(f"INST:CRE:NEW {channel_type}, '{channel_name}'")
+
+    def show_channels(self):
+        raw_list = self.values("INST:LIST?")
+        processed_dict = self._channel_list_to_dict(raw_list)
+        return processed_dict
+
+    def delete_channel(self, channel_name):
+        channels = self.show_channels()
+        strict_discrete_set(channel_name, list(channels.keys()))
+        self.write(f"INST:DEL '{channel_name}'")
+
+    def _channel_list_to_dict(self, raw):
+        d = {key.strip("'"): value.strip("'") for (value, key) in zip(raw[0::2], raw[1::2])}
+        return d
+
+    def select_channel_type(self, application):
```

Comment:
Currently, I do not really understand how to switch the currently active channel or query which channel is active.

Reply:
OK, I found it: page 879 in https://scdn.rohde-schwarz.com/ur/pws/dl_downloads/pdm/cl_manuals/user_manual/1173_9411_01/FSW_UserManual_en_49.pdf

It states that you can also select the Channel by `<ChannelName>`. I would suggest that we do not allow selecting of a channel by `<ChannelType>`, since this can be ambiguous if there are multiple channels with the same type.

You should check whether you can query the active channel with `INST?`. If so, I would suggest that you define a `Instrument.control` named `active_channel` where the active channel can be set and queried. You could use the validator `strict_discrete_set` with the output of `show_channels`. However, dynamic `values` for validators do not work when using the `Instrument.control`. Thus, I suggest to manually creating properties with `@property`.